azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -0,0 +1,378 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	sqlParse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sql/parse"
+	sqlValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/sql/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/synapse/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/synapse/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmSynapseSqlPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmSynapseSqlPoolCreate,
+		Read:   resourceArmSynapseSqlPoolRead,
+		Update: resourceArmSynapseSqlPoolUpdate,
+		Delete: resourceArmSynapseSqlPoolDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				if _, err := parse.SynapseSqlPoolID(d.Id()); err != nil {
+					return []*schema.ResourceData{d}, err
+				}
+
+				d.Set("create_mode", "Default")
+				if v, ok := d.GetOk("create_mode"); ok && v.(string) != "" {
+					d.Set("create_mode", v)
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SynapseSqlPoolName,
+			},
+
+			"synapse_workspace_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SynapseWorkspaceID,
+			},
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "DW1000c",
+				ValidateFunc: validation.StringInSlice([]string{
+					"DW100c",
+					"DW200c",
+					"DW300c",
+					"DW400c",
+					"DW500c",
+					"DW1000c",
+					"DW1500c",
+					"DW2000c",
+					"DW2500c",
+					"DW3000c",
+					"DW5000c",
+					"DW6000c",
+					"DW7500c",
+					"DW10000c",
+					"DW15000c",
+					"DW30000c",
+				}, false),
+			},
+
+			"create_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Default",
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Default",
+					"Recovery",
+					"PointInTimeRestore",
+				}, false),
+			},
+
+			"collation": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"source_database_id": {
```

Comment:
Would this make more sense as 
```suggestion
			"restore_source_database_id": {
```

Reply:
according to the service team, sql pool backend service is a proxy to sql database. In the azurerm_sql_database, the field is `source_database_id`, so I think we would better keep them the same ?